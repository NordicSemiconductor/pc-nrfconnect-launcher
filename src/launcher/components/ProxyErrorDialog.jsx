/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React from 'react';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import { bool, func } from 'prop-types';

const ProxyErrorDialog = ({ isVisible, onOk }) => (
    <Modal show={isVisible} backdrop>
        <Modal.Header>
            <Modal.Title>Proxy error</Modal.Title>
        </Modal.Header>
        <Modal.Body>
            <p>
                It appears that you are having problems authenticating with a
                proxy server. This will prevent you from using certain features
                of nRF Connect, such as installing apps from the
                &quot;Add/remove apps&quot; screen.
            </p>
            <p>
                If you are unable to resolve the issue, then go to Settings and
                disable &quot;Check for updates at startup&quot;. Then restart
                nRF Connect and install apps manually by following the
                instructions at
                https://github.com/NordicSemiconductor/pc-nrfconnect-launcher.
            </p>
        </Modal.Body>
        <Modal.Footer>
            <Button onClick={onOk}>OK</Button>
        </Modal.Footer>
    </Modal>
);

ProxyErrorDialog.propTypes = {
    isVisible: bool.isRequired,
    onOk: func.isRequired,
};

export default ProxyErrorDialog;
