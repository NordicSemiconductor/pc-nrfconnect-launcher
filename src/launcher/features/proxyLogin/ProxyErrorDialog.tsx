/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React from 'react';
import {
    ErrorDialog,
    ExternalLink,
} from '@nordicsemiconductor/pc-nrfconnect-shared';

import { useLauncherDispatch, useLauncherSelector } from '../../util/hooks';
import { getIsErrorVisible, loginErrorDialogClosed } from './proxyLoginSlice';

export default () => {
    const dispatch = useLauncherDispatch();
    const isVisible = useLauncherSelector(getIsErrorVisible);

    return (
        <ErrorDialog
            isVisible={isVisible}
            onHide={() => dispatch(loginErrorDialogClosed())}
            title="Proxy error"
        >
            <p>
                It appears that you are having problems authenticating with a
                proxy server. This will prevent you from using certain features
                of nRF Connect for Desktop, such as installing downloadable apps
                or checking for updates.
            </p>
            <p>
                If you are unable to resolve the issue, then go to Settings and
                disable &quot;Check for updates at startup&quot;. Then restart
                nRF Connect for Desktop and install apps manually by following
                the instructions at{' '}
                <ExternalLink href="https://github.com/NordicSemiconductor/pc-nrfconnect-launcher" />
                .
            </p>
        </ErrorDialog>
    );
};
