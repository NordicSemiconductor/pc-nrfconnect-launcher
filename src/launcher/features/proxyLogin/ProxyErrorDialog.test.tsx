/*
 * Copyright (c) 2022 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React from 'react';

import render from '../../../test/testrenderer';
import ProxyErrorDialog from './ProxyErrorDialog';
import {
    loginCancelledByUser,
    loginErrorDialogClosed,
    loginRequestedByServer,
} from './proxyLoginSlice';

const requestId = 'test ID';
const authInfo = {
    isProxy: true,
    scheme: 'https',
    host: 'host',
    port: 4711,
    realm: 'realm',
};

describe('ProxyErrorDialog', () => {
    it('is initially invisible', () => {
        expect(
            render(<ProxyErrorDialog />, [
                loginRequestedByServer({ requestId, authInfo }),
            ]).baseElement
        ).toMatchSnapshot();
    });

    it('is displayed after users canceled the login', () => {
        expect(
            render(<ProxyErrorDialog />, [
                loginRequestedByServer({ requestId, authInfo }),
                loginCancelledByUser(),
            ]).baseElement
        ).toMatchSnapshot();
    });

    it('is hidden again after users confirmed the login', () => {
        expect(
            render(<ProxyErrorDialog />, [
                loginRequestedByServer({ requestId, authInfo }),
                loginCancelledByUser(),
                loginErrorDialogClosed(),
            ]).baseElement
        ).toMatchSnapshot();
    });
});
