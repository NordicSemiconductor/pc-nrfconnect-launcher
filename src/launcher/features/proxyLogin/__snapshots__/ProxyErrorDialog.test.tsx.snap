// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProxyErrorDialog is displayed after users canceled the login 1`] = `
<body
  class="modal-open"
  style="padding-right: 0px;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal show"
    data-focus-visible-added=""
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title h4"
          >
            Proxy error
          </div>
        </div>
        <div
          class="modal-body"
        >
          <p>
            It appears that you are having problems authenticating with a proxy server. This will prevent you from using certain features of nRF Connect for Desktop, such as installing apps from the "Add/remove apps" screen.
          </p>
          <p>
            If you are unable to resolve the issue, then go to Settings and disable "Check for updates at startup". Then restart nRF Connect for Desktop and install apps manually by following the instructions at https://github.com/NordicSemiconductor/pc-nrfconnect-launcher.
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            OK
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ProxyErrorDialog is hidden again after users confirmed the login 1`] = `
<body
  class=""
  style="padding-right: 0px;"
>
  <div />
</body>
`;

exports[`ProxyErrorDialog is initially invisible 1`] = `
<body>
  <div />
</body>
`;
